### PR TITLE
attempt to do better padding on pieces being written into sectors

### PR DESCRIPTION
**Pull request by whyrusleeping**

### CODE REVIEW

**Comment by ribasushi** [comment]
Relevant PR's to make this all work:
```
diff --git a/go.mod b/go.mod
index 503e95c5d..2d2548ccc 100644
--- a/go.mod
+++ b/go.mod
@@ -31 +31 @@ require (
-       github.com/filecoin-project/go-commp-utils v0.0.0-20201119054358-b88f7a96a434
+       github.com/filecoin-project/go-commp-utils v0.1.1-0.20210407220142-9d7f7aaf8c9c
@@ -35 +35 @@ require (
-       github.com/filecoin-project/go-fil-markets v1.1.9
+       github.com/filecoin-project/go-fil-markets v1.1.10-0.20210407195421-65658e33ed64
```

https://github.com/filecoin-project/go-fil-markets/pull/536
https://github.com/filecoin-project/go-commp-utils/pull/3

**Review by ribasushi — APPROVED** [review]
This has been tested extensively on nerpanet:

- Storage of an 4x artificially padded data filling entire sector
- Storage of a 2x artificially padded data filling half of sector, with the other half being auto-filled by CC
- Retrieval from a separate node from the saved-by-default unsealed copy
- Retrieval from a separate node while triggering unsealing on the miner

:shipit:

**Review by magik6k — COMMENTED** [review]
Some tests would be nice (sealer_test.go, should be easy enough to set them up)

File: extern/sector-storage/ffiwrapper/sealer_cgo.go [hunk 1]
```diff
@@ -195,19 +195,32 @@ func (sb *Sealer) AddPiece(ctx context.Context, sector storage.SectorRef, existi
...
 
+	if payloadRoundedBytes < pieceSize.Padded() {
+		paddedCid, err := commpffi.ZeroPadPieceCommitment(pieceCID, payloadRoundedBytes.Unpadded(), pieceSize)
+		if err != nil {
+			return abi.PieceInfo{}, xerrors.Errorf("failed to pad data: %w", err)
+		}
+
+		pieceCID = paddedCid
+	}
+
```
Comment (magik6k) [inline extern/sector-storage/ffiwrapper/sealer_cgo.go:223]: Should be below the `if .. commcid.CIDToPieceCommitmentV1` check

**Review by magik6k — COMMENTED** [review]
1 Nit

**Review by magik6k — APPROVED** [review]

**Review by magik6k — APPROVED** [review]

**Pull request merged**